weld/jpa/src/main/java/org/jboss/as/weld/services/bootstrap/WeldJpaInjectionServices.java

```diff
@@ -122,6 +124,15 @@ private String getScopedPUName(final DeploymentUnit deploymentUnit, final String
         return scopedPu.getScopedPersistenceUnitName();
     }
 
+    private static Map getProperties(PersistenceContext context) {
+        HashMap map = new HashMap();
```

Comment:
If TransactionScopedEntityManager cached this map in its 'properties' field that wouldn't be thread safe.

Reply:
More about thread safety:

As per https://jakarta.ee/specifications/persistence/3.2/jakarta-persistence-spec-3.2#obtaining-an-entitymanager

> An entity manager must not be shared among multiple concurrently executing threads, as the entity manager and persistence context are not required to be threadsafe. Entity managers must only be accessed in a single-threaded manner.